File: frontend/event-finder-frontend/src/app/dtos/event.ts
export interface EventOrganizer {
    id: string;
    name: string;
    avatarUrl?: string;
}

export interface EventTag {
    id: string;
    name: string;
}

export interface EventRegistration {
    id: string;
    userId: string;
    userName: string;
    registeredAt: string;
}

export interface Event {
    id: string;
    title: string;
    description: string;
    date: string;
    time: string;
    location: string;
    address: string;
    image: string;
    category: string;
    totalSpots: number;
    availableSpots: number;
    organizerId: string;
    organizer: EventOrganizer;
    tags: EventTag[];
    registrations: EventRegistration[];
    createdAt: string;
    updatedAt: string;
}

export interface CreateEventData {
    title: string;
    description: string;
    date: string;
    time: string;
    location: string;
    address: string;
    image: string;
    category: string;
    totalSpots: number;
    tagIds?: string[];
}

export interface UpdateEventData {
    title?: string;
    description?: string;
    date?: string;
    time?: string;
    location?: string;
    address?: string;
    image?: string;
    category?: string;
    totalSpots?: number;
    tagIds?: string[];
}

File: backend/EventFinder/EventFinder.Domain/Entities/Event.cs
namespace EventFinder.Domain.Entities
{
    public class Event : Entity
    {
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string Date { get; set; } = null!;           // "2026-03-22"
        public string Time { get; set; } = null!;           // "18:00"
        public string Location { get; set; } = null!;
        public string Address { get; set; } = null!;
        public double[] Coordinates { get; set; } = null!;  // [lat, lon]
        public string Category { get; set; } = null!;
        public string? Image { get; set; }
        public string[]? Images { get; set; }
        public string OrganizerId { get; set; }
        public string OrganizerName { get; set; } = null!;
        public string? OrganizerAvatar { get; set; }
        public int? AvailableSpots { get; set; }
        public int? TotalSpots { get; set; }

        //public long TagId { get; set; }

        //public Tag Tag { get; set; } = null!;

        public List<Registration> Registrations { get; set; }
    }
}
